routing/dht/routing.go

```diff
@@ -28,6 +29,8 @@ var asyncQueryBuffer = 10
 // PutValue adds value corresponding to given Key.
 // This is the top level "Store" operation of the DHT
 func (dht *IpfsDHT) PutValue(ctx context.Context, key key.Key, value []byte) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*30)
```

Comment:
still not sure how i feel about this. I think the callers should be able to specify how long this should take.

@jbenet thoughts on this?